shell/src/config.rs

```diff
@@ -4,9 +4,10 @@ use std::path::{Path, PathBuf};
 
 #[derive(Debug, Clone)]
 pub struct ShellConfig {
+    pub base_path: PathBuf,
     pub keystore: KeystoreConfig,
     pub subxt: SubxtConfig,
-    pub bind_ip: String,
+    pub bind_ip: std::net::Ipv4Addr,
```

Comment:
We should generalize this to an `IpAddr` to support both IPv6 and IPv4. IPv6 is very NAT traversal friendly.

Reply:
Good point.